src/nvim/eval.lua

```diff
@@ -10459,7 +10460,8 @@ M.funcs = {
     ]=],
     fast = true,
     name = 'stdpath',
-    params = { { 'what', 'any' } },
+    params = { { 'what', 'string' } },
```

Comment:
Can we go one step further and have `what` be an exhaustive union of `cache|config|...`, or will this cause issue for generating docs?

Reply:
I would certainly like that, but that's a question for an experienced maintainer. I'm not sure of how "narrow" these generated types are allowed to be.

Reply:
`'cache'|'config'|...` should work